webapp/app/js/filters.js

```diff
@@ -39,11 +39,26 @@ treeherder.filter('stripHtml', function() {
 
 treeherder.filter('linkifyBugs', function() {
     return function(input) {
-        var re = new RegExp('(?:Bug (\\d+))', 'ig');
-        var str = input || '';
-        return str.replace(re,
-            '<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=$1" target="_blank" title="Bug $1">$1</a>'
-        );
+        if (input) {
+            var re = /.*-- ([0-9]+)|.*bug-([0-9]+)|.*Bug ([0-9]+)/ig; //Matches 3 paterrns "<-- XXX" , "bug-XXX" and "Bug XXX"
```

Comment:
Let's move the comment above (also, typo: paterrns -> patterns) and just execute that raw regex, instead of creating a temporary variable re.